### PR TITLE
Issue #13345: Enable examples tests for RecordComponentNameCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for RequireThisCheck

**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **RecordComponentNameCheck** in the project. The implementation focuses on ensuring that RecordComponentName checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/recordcomponentname/Example2.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+</module>
+*/
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.recordcomponentname;
+
+// xdoc section -- start
+class Example2 {
+  record MyRecord1(String value, int other) {}
+
+  record MyRecord2(String... strings) {}
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/recordcomponentname/Example2.java:17]: Please keep examples the same.
This record is absent in example1.

To spot it yourself , please generate website, and copy content of examples to some diff text tool, there are few free in web.
Looks at issue description images for example.

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/recordcomponentname/Example2.java (hunk 1 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/recordcomponentname/Example2.java]: Done

**Comment by romani** [comment]
Thanks a lot !!!!

**Pull request merged**